lib/Backend/LinearScan.cpp

```diff
@@ -1775,8 +1775,12 @@ LinearScan::FillBailOutRecord(IR::Instr * instr)
             uint outParamCount = bailOutInfo->GetStartCallOutParamCount(i);
             startCallOutParamCounts[i] = outParamCount;
 #ifdef _M_IX86
-            uint orphanedArgCount = 0;
-            uint deadArgCount = 0;
+            if (bailOutInfo->startCallInfo[i].instr->m_opcode == Js::OpCode::StartCall)
+            {
+                // Deadcode might have eliminated the argouts and the call instruction due to a BailOnNoProfile after StartCall
+                // In such cases, StartCall opcode is not changed to LoweredStartCall, mark the StartCall instruction accordingly
+                bailOutInfo->startCallInfo[i].isOrphanedCall = true;
```

Comment:
Do we even need the bool `isOrphanedCall` on `StartCallInfo` then?

Reply:
We use it later on in Lowerer::EmitEHBailoutStackRestore.